light_elf.h

```diff
@@ -0,0 +1,98 @@
+/* This is a derivative work of https://github.com/torvalds/linux/blob/master/include/uapi/linux/elf.h */
+
+#ifndef _ELF_H
+#define _ELF_H 1
+
+#include <inttypes.h>
+
+__BEGIN_DECLS
+
+typedef uint16_t Elf32_Half;
+typedef uint16_t Elf64_Half;
+typedef uint32_t Elf32_Word;
+typedef uint32_t Elf64_Word;
+typedef uint64_t Elf64_Xword;
+typedef uint32_t Elf32_Addr;
+typedef uint64_t Elf64_Addr;
+typedef uint32_t Elf32_Off;
+typedef uint64_t Elf64_Off;
+
+#define EI_NIDENT 16
+
+typedef struct elf32_hdr {
+    unsigned char e_ident[EI_NIDENT];
+    Elf32_Half e_type;
+    Elf32_Half e_machine;
+    Elf32_Word e_version;
+    Elf32_Addr e_entry; /* Entry point */
+    Elf32_Off e_phoff;
+    Elf32_Off e_shoff;
+    Elf32_Word e_flags;
+    Elf32_Half e_ehsize;
+    Elf32_Half e_phentsize;
+    Elf32_Half e_phnum;
+    Elf32_Half e_shentsize;
+    Elf32_Half e_shnum;
+    Elf32_Half e_shstrndx;
+} Elf32_Ehdr;
+
+typedef struct elf64_hdr {
+    unsigned char e_ident[EI_NIDENT]; /* ELF "magic number" */
+    Elf64_Half e_type;
+    Elf64_Half e_machine;
+    Elf64_Word e_version;
+    Elf64_Addr e_entry; /* Entry point virtual address */
+    Elf64_Off e_phoff; /* Program header table file offset */
+    Elf64_Off e_shoff; /* Section header table file offset */
+    Elf64_Word e_flags;
+    Elf64_Half e_ehsize;
+    Elf64_Half e_phentsize;
+    Elf64_Half e_phnum;
+    Elf64_Half e_shentsize;
+    Elf64_Half e_shnum;
+    Elf64_Half e_shstrndx;
+} Elf64_Ehdr;
+
+typedef struct elf32_shdr {
+    Elf32_Word sh_name;
+    Elf32_Word sh_type;
+    Elf32_Word sh_flags;
+    Elf32_Addr sh_addr;
+    Elf32_Off sh_offset;
+    Elf32_Word sh_size;
+    Elf32_Word sh_link;
+    Elf32_Word sh_info;
+    Elf32_Word sh_addralign;
+    Elf32_Word sh_entsize;
+} Elf32_Shdr;
+
+typedef struct elf64_shdr {
+    Elf64_Word sh_name; /* Section name, index in string tbl */
+    Elf64_Word sh_type; /* Type of section */
+    Elf64_Xword sh_flags; /* Miscellaneous section attributes */
+    Elf64_Addr sh_addr; /* Section virtual addr at execution */
+    Elf64_Off sh_offset; /* Section file offset */
+    Elf64_Xword sh_size; /* Size of section in bytes */
+    Elf64_Word sh_link; /* Index of another section */
+    Elf64_Word sh_info; /* Additional section information */
+    Elf64_Xword sh_addralign; /* Section alignment */
+    Elf64_Xword sh_entsize; /* Entry size if section holds table */
+} Elf64_Shdr;
+
+/* Note header in a PT_NOTE section */
+typedef struct elf32_note {
+    Elf32_Word n_namesz; /* Name size */
+    Elf32_Word n_descsz; /* Content size */
+    Elf32_Word n_type; /* Content type */
+} Elf32_Nhdr;
+
+#define ELFCLASS32  1
+#define ELFDATA2LSB 1
+#define ELFDATA2MSB 2
+#define ELFCLASS64  2
+#define EI_CLASS    4
+#define EI_DATA     5
+
+__END_DECLS
+
+#endif /* elf.h */
```

Comment:
This file is still licensed as GNU GPL. This license is not compatible with the rest of the code base. This is a really big problem, and we cannot risk a GPL violation by relicensing the extracted code. You will have to write it yourself, I guess, and change the variable and macro names etc. The file doesn't have to be compatible to any other code any more, especially since this is going to be used on both Linux and OS X.